src/main/java/com/exasol/adapter/request/parser/PushdownSqlParser.java

```diff
@@ -420,12 +430,27 @@ private SqlNode parsePredicateBetween(final JsonObject exp) {
         return new SqlPredicateBetween(betweenExp, betweenLeft, betweenRight);
     }
 
+    private SqlNode parsePredicateIsJson(final JsonObject jsonExpression) {
+        final SqlNode expression = parseExpression(jsonExpression.getJsonObject(EXPRESSION));
+        final SqlPredicateIsJson.TypeConstraints typeConstraint = SqlPredicateIsJson.TypeConstraints
+                .valueOf(jsonExpression.getString("typeConstraint").toUpperCase());
+        final SqlPredicateIsJson.KeyUniquenessConstraint keyUniquenessConstraint = SqlPredicateIsJson.KeyUniquenessConstraint
+                .of(jsonExpression.getString("keyUniquenessConstraint"));
+        return new SqlPredicateIsJson(expression, typeConstraint, keyUniquenessConstraint);
+    }
+
+    private SqlNode parsePredicateIsNotJson(final JsonObject jsonExpression) {
+        final SqlNode expression = parseExpression(jsonExpression.getJsonObject(EXPRESSION));
+        final SqlPredicateIsJson.TypeConstraints typeConstraint = SqlPredicateIsJson.TypeConstraints
+                .valueOf(jsonExpression.getString("typeConstraint").toUpperCase());
+        final SqlPredicateIsJson.KeyUniquenessConstraint keyUniquenessConstraint = SqlPredicateIsJson.KeyUniquenessConstraint
+                .of(jsonExpression.getString("keyUniquenessConstraint"));
+        return new SqlPredicateIsNotJson(expression, typeConstraint, keyUniquenessConstraint);
```

Comment:
Here maybe using single class (suggested below), the common parts of these two methods can be extracted to another function. And set the predicate variable accordingly in these two methods.